supplier/supplier.cc

```diff
@@ -72,6 +65,23 @@ class SupplierServiceImpl final : public Supplier::Service {
     }
     return Status::OK;
   }
+
+  Status AddVendor(ServerContext* context, const VendorInfo* request,
+                   Empty* info) {
+    if (vendor_count_ > 9) {
```

Comment:
Using a vector for your vendor storage would remove the need for this hard limit of 9 vendors.

Reply:
Yeah I think there might be better workaround, but for the sake of time, I decide to use a fix number of vendors.